### PR TITLE
Add bookmark

### PR DESCRIPTION
**Pull request by alisa-lin**
## Change Log
- Can add bookmarks from the new button
- Bookmark and clink mapping database created
- Bookmark menu code was (unnecessarily) refactored (it will be necessary later but didn't need to be done for this PR oops)
- Bookmarks got a deleted flag 
- Clinks got a privacy toggle & flag
- Note that bookmarks do NOT need to have a unique title/URL

### CODE REVIEW

File: flask-server/main.py [hunk 1]
```diff
@@ -229,7 +228,32 @@ def check_denylist(token):
     if token_query:
         return True
     else:
-        return False
+        return False        
+
+@app.route('/apis/add-bookmark', methods=['POST'])
```
Comment (marsw42) [inline flask-server/main.py:233]: Can add in the future but also we should validate user before adding bookmarks

File: react-app/src/components/Common/bookmarkmenu.js [hunk 2]
```diff
@@ -8,109 +8,102 @@ const layout = {
   layout: 'vertical'
 };
 
-class BookmarkMenu extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { loading: false, visible: false };
-  }
+const BookmarkMenu = () => {
+  const [visible, setVisible] = useState(false);
```
Comment (marsw42) [inline react-app/src/components/Common/bookmarkmenu.js:12]: `isVisible`, `isLoading` for booleans

File: react-app/src/components/Common/bookmarkmenu.js [hunk 3]
```diff
@@ -8,109 +8,102 @@ const layout = {
...
+    setLoading(true);
     setTimeout(() => {
-      this.setState({ loading: false, visible: false });
+      setLoading(false);
+      setVisible(false);
     }, 3000);
   };
 
-  onFinishFailed = errorInfo => {
+  const onFinishFailed = errorInfo => {
```
Comment (marsw42) [inline react-app/src/components/Common/bookmarkmenu.js:27]: Remove `console.log`

File: react-app/src/features/clink.js [hunk 4]
```diff
@@ -41,4 +52,18 @@ export const addClink = (title, id, callbackSucceed, callbackFailed) => async di
   }
 }
 
+export const addBookmark = (link, title, description, clink, callbackSucceed, callbackFailed) => async dispatch => {
+  try {
+    dispatch(addBookmarkStart())
+    const response = await apis.addBookmark(link, title, description, clink)
+    console.log(response)
```
Comment (marsw42) [inline react-app/src/features/clink.js:59]: remove console.log

File: react-app/src/features/clink.js [hunk 5]
```diff
@@ -41,4 +52,18 @@ export const addClink = (title, id, callbackSucceed, callbackFailed) => async di
...
+    dispatch(addBookmarkSucceed())
+    callbackSucceed()
+  } catch (err) {
+    console.log(err)
```
Comment (marsw42) [inline react-app/src/features/clink.js:63]: remove this line

File: flask-server/main.py (hunk 1 above)
Comment (michaeleggboy) [reply flask-server/main.py:233]: But in theory if verify the jwt when the user goes to the page, do we have verify again when in the api

**Pull request merged**